src/data/annoucementData.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  addDoc,
+  collection,
+  Timestamp,
+  doc,
+  getDoc,
+  getDocs,
+  updateDoc,
+  deleteDoc,
+} from "firebase/firestore";
+import { db } from "../firebase";
+
+type AnnouncementType = "info" | "warning" | "error";
+
+interface Announcement {
+  _id?: string;
+  type: AnnouncementType;
+  message: string;
+  createdAt?: Timestamp;
+  updatedAt?: Timestamp;
+  scheduleDate: Timestamp;
+  expirationDate: Timestamp;
+  active: boolean;
+}
+
+export async function createAnnouncement(
```

Comment:
Should we implement, try-catch pattern here? Ik running it in the server returned error but not sure how we want to display them to the users